Auto reconnect in case of connection loss

fillson-shady opened the pull request:
Automatically try to reconnect if connection is interrupted.  
Note that if a queue name is not specified then all messages arrived during reconnecting will be lost.  
If the queue name is set then messages will be store there and delivered after reconnection.

ghokun commented:
This will take me some time to review. Just a quick look, we may need an exponential backoff to retry connection.

fillson-shady commented:
Of course, take your time!  
Regarding the backoff - by default a rabbit client will wait for 30 seconds before retrying. So it won't spam a RabbitMQ cluster.  
On the other hand we can think of lowering this 30 seconds timeout to 1 second for example (to be able to reconnect ASAP). Then the backoff strategy will be required for sure. WDYT?

ghokun commented:
Here is an example here: https://github.com/rabbitmq/amqp091-go/blob/main/_examples/client/client.go

fillson-shady commented:
Hi @ghokun, I performed some refactoring, extracted client to a separate file and added a simple backoff strategy to reconnection logic. Thank you for the provided example, it helped a lot!

ghokun commented on client.go:
```diff
@@ -0,0 +1,349 @@
+	routingKey string
+}
+
+type RabbitMQCConsumer struct {
```
```suggestion
type RabbitMQConsumer struct {
```
